api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -1,18 +1,242 @@
+        .expect(
+          200,
+          collectionEnvelope(emptyAssessmentsSummaryList, 0),
+          err => {
+            // call a (mock) function that gets a list of programs the user is enrolled in
+            expect(mockPrincipalEnrolledPrograms).toHaveBeenCalledWith(
+              unenrolledPrincipalId
+            );
+            done(err);
+          }
+        );
+    });
+
+    it('should respond with a list of all assessments (without questions) for participant enrolled in one program', done => {
+      mockPrincipalId(enrolledParticipantPrincipalId);
+      const participantAssessmentListResponse = [
+        {
+          curriculum_assessment: curriculumAssessment,
+          program_assessment: programAssessment,
+          submissions_summary: assessmentSubmissionsSummary,
+        },
+      ];
+      // mock response from (function that gets a list of programs the user is enrolled in) to include one program
+      mockPrincipalEnrolledPrograms.mockResolvedValue([4]);
+      // mock response from (call a function that returns the permission of the user for each program (participant/facilitator)) to respond with participant for that one program
+      mockGetProgramIdByProgramAssessmentId.mockResolvedValue([
+        exampleAssessmentId,
+      ]);
+      mockFindRoleInProgram.mockResolvedValue({ title: 'participant' });
+      // mock response from (get a list of program assessments for each program the user is enrolled in) to respond with a list of program assessments
+      mockGetAssessmentsForProgram.mockResolvedValue(programAssessment);
+      // mock responses from (get a list of curriculum assessments that correspond to each program assessment) to respond with the corresponding curriculum assessment for each program assessment in that previous list
+      mockGetCurriculumAssessmentById.mockResolvedValue(curriculumAssessment);
+      // mock responses from (call a (mock) function that gets the participant assessment summary for each program assessment where the user is a participant of that program) to respond with the assessment summary for every program assessment that participant has submitted for previously
+      mockGetAssessmentSubmissionsSummary.mockResolvedValue(
+        assessmentSubmissionsSummary
+      );
+      appAgent
+        .get('/')
+        .expect(
+          200,
+          collectionEnvelope(
+            participantAssessmentListResponse,
+            participantAssessmentListResponse.length
+          ),
+          err => {
+            // call a (mock) function that gets a list of programs the user is enrolled in
+            expect(mockPrincipalEnrolledPrograms).toHaveBeenCalledWith(
+              enrolledParticipantPrincipalId
+            );
+            // call a function that returns the permission of the user for each program (participant/facilitator)
+            expect(mockGetProgramIdByProgramAssessmentId).toHaveBeenCalledWith(
+              programAssessmentId
```

Comment:
The only information you would have access to by the time the flow of the test gets to line 148 would be the following:

- `principalId` of the user making the request
- the return value from your first function (`principalEnrolledPrograms`), which you indicate from line 121 is `[4]`

So, how can we call `getProgramIdByProgramAssessmentId` if neither of these things we have access to include the `programAssessmentId`?